transformer_lens/hook_points.py

```diff
@@ -126,13 +166,15 @@ def clear_context(self):
         del self.ctx
         self.ctx = {}
 
-    def forward(self, x):
+    # TODO: can we assume this will always be a torch.Tensor?
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
         return x
 
     def layer(self):
         # Returns the layer index if the name has the form 'blocks.{layer}.{...}'
         # Helper function that's mainly useful on HookedTransformer
         # If it doesn't have this form, raises an error -
+        assert self.name is not None
```

Comment:
Assertions are just for testing (they are completely ignored by Python if optimization is used (-O)).

The better thing to do here is an if statement, and then raise an error.